### DISCUSSION TITLE
Buy nft but error invalidSigner => 0x815e1d64

### DISCUSSION THREAD

**Discussion by jecombe** [ProjectOpenSea/seaport#973] [Q&A] [answered]
i want buy nft with fullFillBasicOrder. I get order existing with opensea-js like this: 
`{
  considerationToken: '0x0000000000000000000000000000000000000000',
  considerationIdentifier: '0',
  considerationAmount: '250000000000000',
  offerer: '0x1139571D38d7A8199236F62dd58bb7B1256Ee959',
  zone: '0x0000000000000000000000000000000000000000',
  offerToken: '0x317a8Fe0f1C7102e7674aB231441E485c64c178A',
  offerIdentifier: '447973',
  offerAmount: 1,
  basicOrderType: 0,
  startTime: 1674643597,
  endTime: 1677321997,
  zoneHash: '0x0000000000000000000000000000000000000000000000000000000000000000',
  salt: '0x360c6ebe00000000000000000000000000000000000000004fe15af03f9477a3',
  offererConduitKey: '0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000',
  fulfillerConduitKey: '0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000',
  totalOriginalAdditionalRecipients: 2,
  additionalRecipients: [
    {
      amount: '9750000000000000',
      recipient: '0x1139571D38d7A8199236F62dd58bb7B1256Ee959'
    },
    {
      amount: '250000000000000',
      recipient: '0x0000a26b00c1F0DF003000390027140000fAa719'
    }
  ],
  signature: '0xb48bbca9d9350b5bb7f734b7bf5b8c819dd96194fadf74278787fa51ca3f5c2f175eb1c941d4e4af250811b29d05ec5fdac22163ec93a89ebe7b80720b32683b1b'
}` 
and i call function fulFillBasicOrder with smart contract like this:
`function buyNft() public payable {
        AdditionalRecipient memory seaportCut = AdditionalRecipient({
            amount: 250000000000000,
            recipient: payable(0x0000a26b00c1F0DF003000390027140000fAa719)
        });
        AdditionalRecipient memory seaportCc = AdditionalRecipient({
            amount: 9750000000000000,
            recipient: payable(0x1139571D38d7A8199236F62dd58bb7B1256Ee959)
        });
        additionalRecipients.push(seaportCc);

        additionalRecipients.push(seaportCut);

        BasicOrderParameters memory order = BasicOrderParameters({
            considerationToken: 0x0000000000000000000000000000000000000000,
            considerationIdentifier: 0,
            considerationAmount: 9750000000000000,
            offerer: payable(0x1139571D38d7A8199236F62dd58bb7B1256Ee959), // your address
            zone: 0x0000000000000000000000000000000000000000,
            offerToken: 0x317a8Fe0f1C7102e7674aB231441E485c64c178A, // tokenAddress
            offerIdentifier: 447973,
            offerAmount: 1, // total amount
            basicOrderType: BasicOrderType.ETH_TO_ERC721_FULL_OPEN, // ETH_TO_ERC721_FULL_OPEN
            startTime: 1674643597,
            endTime: 1677321997,
            zoneHash: 0x0000000000000000000000000000000000000000000000000000000000000000,
            salt: 0x360c6ebe00000000000000000000000000000000000000004fe15af03f9477a3,
            offererConduitKey: 0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000,
            fulfillerConduitKey: 0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000,
            totalOriginalAdditionalRecipients: 2,
            additionalRecipients: additionalRecipients,
            signature: hex"b48bbca9d9350b5bb7f734b7bf5b8c819dd96194fadf74278787fa51ca3f5c2f175eb1c941d4e4af250811b29d05ec5fdac22163ec93a89ebe7b80720b32683b1b"
        });
        bytes memory exec = abi.encode(order);

        BasicOrderParameters memory basicComponents = abi.decode(
            exec,
            (BasicOrderParameters)
        );
           SeaportInterface(0x00000000006c3852cbEf3e08E8dF289169EdE581)
            .fulfillBasicOrder{value: order.considerationAmount}(basicComponents);
    }`
When i execute my code in remix-IDE, this function give me error 0x815e1d64 =>  invalidSigner.
Why ? 

**Edit: FIXED**

**Comment by wilson-ang** [comment]
Did you managed to find a fix for this issue?
I am using hardhat and the transactions fails with the same error. However, if when I encode the same parameters and function selector and send transaction using Metamask, the transaction will not fail.

**Reply by jecombe** [reply]
Sorry for time.. did you fix that ? I've fixed and i can explain

**Reply by wilson-ang** [reply, marked as answer]
Yes, I managed to fix it. For me the issue was because I didn't specify the forked chain id.

**Comment by waqaswahid1** [comment]
@jecombe @wilson-ang 
i am getting strange error invalid opcode when i do call seaport calldata that i am fetching from the reservoir sell api through eth mainnet forking. 

`Error: VM Exception while processing transaction: invalid opcode
    at <UnrecognizedContract>.<unknown> (0x0000000000000068f116a894984e2db1123eb395)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at EthModule._estimateGasAction (node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:439:7)
    at HardhatNetworkProvider.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:124:18)
    at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)`
